examples/step-with-tasks.yaml

```
@@ -0,0 +1,57 @@
+---
+- step:
+    name: run training
+    image: tensorflow/tensorflow
+    command: python -u /tmp/code/my_mnist.py
+    inputs:
+      - name: training-set-images
+        default: http://yann.lecun.com/exdb/mnist/train-images-idx3-ubyte.gz
+        description: the image data for the training set
+    parameters:
+      - name: B
+        type: integer
+        default: 1
+
+- task:
+    step: run training
+    name: task 1 run training
+    type: bayesian_tpe
+    parameters:
+      - name: A
+        style: single
+        rules:
+          value: 25
```

This is the technical format we use, but I'm not sure if it's necessary to have rules object separated and could be flatten to parameter level? Everyone saves typing.
```suggestion
    parameters:
      - name: A
        style: single
        value: 25
```

---

imho, it will be much harder to validate => more chance to make a mistake. I planned to write schemas for all rules, but    it i s too complex I am open to suggestions :)

---

Do you have plans to match `rules` per `style`? So that it only accepts specific `rules` format when `style` is "single"? I think that should be as easy or even easier when you flat the structure with [oneOf](https://json-schema.org/understanding-json-schema/reference/combining.html#oneof) and having every rule in separated collection.